dashboard/app/controllers/home_controller.rb

```diff
@@ -64,4 +84,29 @@ def terms_and_privacy
   def teacher_announcements
     render template: 'api/teacher_announcement', layout: false
   end
+
+  def recent_courses
+    current_user.in_progress_and_completed_scripts.map do |script|
+      script_id = script[:script_id]
+      script_name = Script.find_by_id(script_id)[:name]
+      {
+        id: script_id,
+        script_name: script_name,
+        courseName: data_t_suffix('script.name', script_name, 'title'),
+        description: data_t_suffix('script.name', script_name, 'description_short'),
+        link: script_url(script_id),
+        image: "",
+        assignedSections: []
+      }
+    end
+  end
+
+  def courses
+    if current_user
+      @recent_courses = recent_courses
```

Comment:
I wonder if you (or anyone else) can explain to me why we have this frequently used pattern of sticking things in an instance variable, and then depending on that instance variable being defined in the view. It makes it much harder to track what's happening with data in my opinion.

The pattern I've taken to is to instead get the data I need before I render a view, pass it to the view using locals, and then extract it using local_assigns.

`</rant>`